certora/specs/AssetsAccounting.spec

```diff
@@ -64,22 +68,24 @@ rule withdrawAssetsAccounting(env e, MorphoHarness.MarketParams marketParams, ui
     assert withdrawnAssets <= ownedAssets;
 }
 
-// Check that the assets borrowed are less than the assets owed in the end.
+// Check that the increase of owed assets are greater than the borrowed assets.
 rule borrowAssetsAccounting(env e, MorphoHarness.MarketParams marketParams, uint256 shares, address onBehalf, address receiver) {
     MorphoHarness.Id id = libId(marketParams);
 
     // Assume no interest as it would increase the total borrowed assets.
     require lastUpdate(id) == e.block.timestamp;
-    // Assume no outstanding debt to begin with.
-    require borrowShares(id, onBehalf) == 0;
+    // Safe require because of the sumBorrowSharesCorrect invariant.
+    require borrowShares(id, onBehalf) <= totalBorrowShares(id);
```

Comment:
not sure why you need to do this, the implem would revert when doing the addition no?

Reply:
Which addition are you talking about ?
We need it here because otherwise you could have much more shares than the total, so supplying artificially increases the assets you are owned (in other words the accounting is broken). But this is a state that can never be reached